src/main/scala/backends/futil/FutilBackend.scala

```diff
@@ -879,7 +879,7 @@ def emitInvokeDecl(app: EApp)(implicit store: Store, id2FuncDef: FunctionMapping
         // the multi-cycle operation is complete, if it exists.
         val (writeEnableSrcPort, delay) = out.multiCycleInfo match {
           case Some((compVar, Some(delay))) =>
-            (CompPort(compVar, "done"), out.delay.map(_ + delay))
+            (CompPort(compVar, "done"), out.delay.map(_ + delay + 1))
```

Comment:
Are we sure that this is always stored in a register? That’s the only time when adding 1 makes sense.

Reply:
Yes, with this `CLet` expression, we're always writing to a register.

https://github.com/cucapra/dahlia/blob/1c88dc75284f13dd11887a20811d7203f0086ee4/src/main/scala/backends/futil/FutilBackend.scala#L870-L871